### ISSUE TITLE
L-BFGS-B not working on Colab / Kaggle with backend tensorflow 2.0

### ISSUE THREAD

**Issue by riccardotomada** [lululxvi/deepxde#969]
As per title, I would like to report that the L-BFGS-B optimizer is currently not working when the selected backend is tensorflow 2.0 on colab / kaggle. As usual, I set the default float type to ‘float64’. What happens is that it gets stuck at the first epoch, the notebook keeps running but nothing is being displayed.

**Comment by elliot-hicks** [comment]
@riccardotomada I had a similar issue, I think its to do with TF's implementation of LBFGS as its the only change from the 1.x to the 2.x backend in DeepXDE. I found easing the [```ftol``` constraint in the LBFGS configuration](https://deepxde.readthedocs.io/en/latest/modules/deepxde.optimizers.html) can stop this issue as it sets when to end an iteration/ epoch (if its too harsh, your first iteration will never conclude). Does that work for you?   Disclaimer - I'm not sure how that will effect the optimisation quality, I haven't delved into the maths.

Also when I moved to the 2.x backend and faced this issue, I found that ADAM is still a good alternative.

**Comment by riccardotomada** [comment]
Thank you very much for the reply @elliot-hicks!

Unfortunately I made some tests and the problem keeps showing, at least until I increase ftol to 1e-6. If I set it to 1e-5 or greater, it just exits at the first epoch.

But I guess is something related to what you suggested.

Yeah Adam is great too and actually my loss is small, in the order of 1e-5. But from my experience L-BFGS-B perform always great after Adam, so I wanted to use it :)
